packages/backend-common/src/scm/git.ts

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import git from 'isomorphic-git';
+import http from 'isomorphic-git/http/node';
+import fs from 'fs-extra';
+import { Logger } from 'winston';
+
+/*
+provider    username  	    password
+GitHub      token	          'x-oauth-basic'
+GitHub App 	token	          'x-access-token'
+BitBucket	  'x-token-auth'	token
+GitLab      'oauth2'	      token
+From : https://isomorphic-git.org/docs/en/onAuth
+
+Azure       'notempty'      token
```

Comment:
Only FYI: Created a PR to update their docs (to add Azure) https://github.com/isomorphic-git/isomorphic-git/pull/1296